### ISSUE TITLE
Add the ability to mark nodes as Favorites

### ISSUE THREAD

**Issue by Jummit** [godotengine/godot-proposals#2922]
### Describe the project you are working on

A large UI program.

### Describe the problem or limitation you are having in your project

When creating large and complicated user interfaces, the scene tree quickly gets cluttered and finding important nodes becomes harder. The search tool can be helpful, but it's still slow to jump between nodes.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Make nodes be able to be added to a Favorite list, simmilar to the file system favorites.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![image](https://user-images.githubusercontent.com/28286961/123509383-75c08100-d675-11eb-92df-de6dd2334dcd.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

It fits well in the core engine.

**Comment by pycbouh** [comment]
Just so I understand you correctly, you mean nodes within scenes? So like a feature for the Scene dock?

**Comment by Jummit** [comment]
> s within scenes? So like a feature for the Scene doc

Correct.

**Comment by Torguen** [comment]
That's right, the scene tree needs improvements in the organizational aspect because when you have a somewhat large level with items, enemies etc ... it is a complete chaos.
I also thought about this proposition but failed to create it.
